cr-examples/triton/src/test/java/oracle/code/triton/TestMatrix.java

```diff
@@ -251,15 +252,15 @@ static void matmul_kernel_broadcast(
             // Load the next block of A and B, generate a mask by checking the K dimension.
             // If it is out of bounds, set it to 0.
             var offs_k_m_0 = expand(offs_k, 0);
-            offs_k_m_0 = compare(offs_k_m_0,
+            var offs_k_m_1 = compare(offs_k_m_0,
                     broadcast(K - k * BLOCK_SIZE_K, offs_k_m_0.type()),
                     LessThan);
-            var a = load(a_ptrs, broadcast(offs_k_m_0, a_ptrs.type()));
-            var offs_k_m_1 = expand(offs_k, 1);
-            offs_k_m_1 = compare(offs_k_m_1,
-                    broadcast(K - k * BLOCK_SIZE_K, offs_k_m_1.type()),
+            var a = load(a_ptrs, broadcast(offs_k_m_1, a_ptrs.type()));
+            var offs_k_m_2 = expand(offs_k, 1);
+            var offs_k_m_3 = compare(offs_k_m_2,
+                    broadcast(K - k * BLOCK_SIZE_K, offs_k_m_2.type()),
                     LessThan);
-            var b = load(b_ptrs, broadcast(offs_k_m_1, b_ptrs.type()));
+            var b = load(b_ptrs, broadcast(offs_k_m_3, b_ptrs.type()));
             // We accumulate along the K dimension.
```

Comment:
Were these necessary changes because you encountered a bug?

Reply:
Because I changed the return type of the `compare` function to `boolean`, It can't match the return type of the `expand` function.